pkg/wormhole/client/health.go

```diff
@@ -0,0 +1,166 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"sync/atomic"
+	"time"
+
+	kitlog "github.com/go-kit/kit/log"
+	"golang.org/x/net/ipv4"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kubernetes_clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/sapcc/kubernikus/pkg/client/kubernetes"
+	"github.com/sapcc/kubernikus/pkg/util/icmp"
+	"github.com/sapcc/kubernikus/pkg/util/netutil"
+)
+
+var sequenceNumber uint32 = 0
+
+const checkInterval = 60 * time.Second
```

Comment:
I wanted to do different recheck intervals at first but in the end didn't because it was to annoying and I didn't want to introduce a state machine. Furthermore the methods from apimachinery `wait` package wouldn't support that so I hat to have done the loop with support for jitter myself. In general I didn't want to overcomplicate the healthcheck at first. If it becomes a problem we can revisit this at a later time.

Reply:
Instead of retrying in different intervals I know added an exponential backoff that retries failed attempts a few times when it worked previously. Hopefully this is enough avoid flakes.

Reply:
👍